File: services/catalog/internal/apps/catalog/dto/service.go
package dto

type CreateServiceVariationDTO struct {
	Name     string  `json:"name"`
	Price    float64 `json:"price"`
	Position int     `json:"position"`
}

type CreateServiceLocationDTO struct {
	LocationType string `json:"location_type"`
	Address      string `json:"address"`
	PhoneMethod  string `json:"phone_method"`
	Phone        string `json:"phone"`
}

type CreateServiceDTO struct {
	Name          string  `json:"name" validate:"required"`
	Description   string  `json:"description"`
	Price         float64 `json:"price"`
	PriceCurrency string  `json:"price_currency"`
	Status        string  `json:"status"`
	ShowInStore   bool    `json:"show_in_store"`

	// Service-specific
	ServiceType      string `json:"service_type"` // appointment | class | course
	Tagline          string `json:"tagline"`
	Duration         string `json:"duration"`
	BufferTime       string `json:"buffer_time"`
	PriceType        string `json:"price_type"` // fixed | variable | free | custom
	PaymentMode      string `json:"payment_mode"`
	CustomPriceLabel string `json:"custom_price_label"`
	BookingMode      string `json:"booking_mode"` // auto | manual

	TagIDs           []string                    `json:"tag_ids"`
	Plans            []CreatePlanDTO             `json:"plans"`
	Variations       []CreateServiceVariationDTO `json:"variations"`
	Locations        []CreateServiceLocationDTO  `json:"locations"`
	AdditionalFields []CreateAdditionalFieldDTO  `json:"additional_fields"`
	MediaKeys        []string                    `json:"media_keys"`
	DisplayImageKey  string                      `json:"display_image_key"`
	DisplayImageID   string                      `json:"display_image_id"`
}

type UpdateServiceDTO = CreateServiceDTO

File: frontend/app/composables/dto/catalog/service.ts
export interface CreateServiceVariationDTO { name: string; price: number; position: number }
export interface CreateServiceLocationDTO { location_type: string; address: string; phone_method: string; phone: string }
export interface CreateAdditionalFieldDTO { label: string; field_type: string; value: string; position: number }

export interface CreatePlanDTO {
  name: string
  plan_type: 'subscription' | 'package'
  price?: number
  price_currency?: string
  billing_cycle?: 'monthly' | 'yearly'
  session_count?: number | null  // null = unlimited
  validity_days?: number | null  // package only
}

export interface Plan {
  id: string
  catalog_item_id: string
  crm_id: string
  name: string
  plan_type: 'subscription' | 'package'
  price: number
  price_currency: string
  billing_cycle?: string
  session_count?: number | null
  validity_days?: number | null
  status: string
  created_at: string
  updated_at: string
}

export interface CreateServiceDTO {
  name: string
  description?: string
  price?: number
  price_currency?: string
  discount_percent?: number
  status?: string
  show_in_store?: boolean
  service_type?: string
  tagline?: string
  duration?: string
  buffer_time?: string
  price_type?: string
  payment_mode?: string
  custom_price_label?: string
  booking_mode?: string
  tag_ids?: string[]
  plans?: CreatePlanDTO[]
  variations?: CreateServiceVariationDTO[]
  locations?: CreateServiceLocationDTO[]
  additional_fields?: CreateAdditionalFieldDTO[]
}

export type UpdateServiceDTO = CreateServiceDTO
